### DISCUSSION TITLE
Audio skipping on iOS

### DISCUSSION THREAD

**Discussion by KupoGames** [airsdk/Adobe-Runtime-Support#2428] [Q&A] [answered]
Hi, I have recently published my app on iOS, and background music occasionally "skips" like a vinyl record. This seems to happen across different devices, such as the iPhone 11 and iPhone 14. It's quite minor, but a little bit annoying.

I have never seen this issue on Windows or Android, even though the same code is used.
I'm just using the SoundChannel class to play mp3 and Wav files, and controlling the volume with SoundTransform. 
Only one music track is played at a time this way, but there are other timeline animation sound effects too.

The app is free-to-play on the App Store, if you want to try to reproduce it - although it may be hard to notice if you're not familiar with the music:
https://apps.apple.com/mm/app/epic-battle-fantasy-5-rpg/id6443990313

Did I do something wrong, or is this a bug with AIR/Flash on iOS?
Thanks.

Unrelated: I gotta say, Flash runs great on iOS! Much smoother than any other platform. Very cool.

**Comment by ajwfrost** [comment, marked as answer]
Hi

Just to check whether this is the same as issue that we're currently working on (which is much more prevalent on iPhone 14) - is it like the audio just skips briefly? So it's not jumping forward/backward in the track, it's just like a brief pause is inserted for a split second?

We've got a reproduction of this by generating a sound wave using sample data, the current belief is that it's a threading issue where the OS is running out of data before we're able to provide a new buffer.

thanks

**Reply by KupoGames** [reply]
Hmm, that sounds like it's probably the same issue - it's only iPhone 14 users that have mentioned it to me.
I noticed it on my iPhone 11 before, but I'm testing it now, and it's not happening this time. 

Thanks, keep up the good work.
